src/fight/mod.rs

```diff
@@ -3,157 +3,280 @@
+fn evaluate_gauss_interval(
+    start: usize,
+    end: usize,
+    gauss_len: usize,
+    success_probability: f64,
+) -> f64 {
+    math_tools::sigma(
+        start as isize,
+        end as isize,
+        |current, params, _, _| {
+            math_tools::compute_bernoulli(params.unwrap().0, current as usize, params.unwrap().1)
+        },
+        Some((gauss_len, success_probability)),
+    )
 }
 
-/// Compute the first round of a fight between 2 Regiment
+/// Compute the average damage dealt by a unit to another according to the requested scenario
 ///
-/// # Parameters
-/// regiment_attacking (&regiment::Regiment): The Regiment attacking
+/// ## Parameters
+/// (&regiment::Regiment) attacking_regiment: The attacker
 ///
-/// regiment_defending (&regiment::Regiment): The Regiment defending
+/// (&regiment::Regiment) defending_regiment: The defender
 ///
-/// # Return
-/// usize: The number of wounds the defending Regiment will suffer
-fn fight_first_turn(
-    regiment_attacking: &regiment::Regiment,
-    regiment_defending: &regiment::Regiment,
-) -> usize {
-    let regiment_attacking_stats: &model::Stats = regiment_attacking.get_model().get_stats();
-    let regiment_defending_stats: &model::Stats = regiment_defending.get_model().get_stats();
-    let to_hit: usize = roll::compute_roll_to_hit(
-        regiment_attacking_stats.get_offensive(),
-        regiment_defending_stats.get_defense(),
-    );
-    let to_wound: usize = roll::compute_roll_to_wound(
-        regiment_attacking_stats.get_strength(),
-        regiment_defending_stats.get_resilience(),
+/// (ComputeCase) case: The scenario from first_unit point of view
+///
+/// ## Return
+/// (usize, f64): The average amount of damage dealt by first_unit and the probability for this scenario to occurs
+fn compute_case(
+    attacking_regiment: &regiment::Regiment,
+    defending_regiment: &regiment::Regiment,
+    case: &ComputeCase,
+) -> (usize, f64) {
+    let attacking_stats: &model::Stats = attacking_regiment.get_model().get_stats();
+    let nb_touch: usize =
+        (attacking_stats.get_attack() as f64 * 1.5 * attacking_regiment.get_cols() as f64).round()
+            as usize;
+    let wound_probability: f64 = computation_tools::compute_damage_probability(
+        attacking_stats,
+        defending_regiment.get_model().get_stats(),
     );
-    let nb_attacks: usize = get_nb_attacks(regiment_attacking);
-    let nb_wounds: usize = get_nb_wounds(nb_attacks, to_hit, to_wound);
-    let final_result: usize = get_final_result(
-        nb_wounds,
-        regiment_defending_stats.get_armour(),
-        regiment_attacking_stats.get_armour_penetration(),
+    let max_hit: usize = std::cmp::min(nb_touch, defending_regiment.get_regiment_health_points());
+    let checkpoints: (usize, usize) = find_great_gauss_checkpoints(
+        nb_touch,
+        wound_probability,
+        defending_regiment.get_regiment_health_points(),
```

Comment:
Maybe store the return value of `get_regiment_health_points()` in a variable to avoid another jump